cmd/internal/server/server.go

```diff
@@ -138,34 +138,50 @@ func (c *connectorServer) Update(request *fivetran_sdk.UpdateRequest, server fiv
 		return status.Error(codes.InvalidArgument, "request did not contain a valid selection")
 	}
 
-	schema, ok := request.Selection.Selection.(*fivetran_sdk.Selection_WithSchema)
+	schemaSelection, ok := request.Selection.Selection.(*fivetran_sdk.Selection_WithSchema)
 	if !ok {
 		return status.Error(codes.InvalidArgument, "request did not contain a Selection_WithSchema")
 	}
 
-	logger := handlers.NewLogger(server, requestId, psc.TreatTinyIntAsBoolean)
-	defer logger.Release()
-	var db lib.ConnectClient
-	if c.clientConstructor == nil {
-		mysql, err := lib.NewMySQL(psc)
+	var (
+		db          lib.ConnectClient
+		mysqlClient lib.MysqlClient
+	)
+
+	if c.mysqlClientConstructor == nil {
+		mysqlClient, err = lib.NewMySQL(psc)
 		if err != nil {
 			return status.Error(codes.InvalidArgument, "unable to open connection to PlanetScale database")
 		}
-		defer mysql.Close()
-		db = lib.NewConnectClient(&mysql)
+		defer mysqlClient.Close()
```

Comment:
Move the defer close below the conditional block so that a custom constructed client also gets closed.